ibm/resource_ibm_container_cluster.go

```diff
@@ -771,6 +779,8 @@ func resourceIBMContainerClusterRead(d *schema.ResourceData, meta interface{}) e
 	var poolName string
 	var poolContains bool
 
+	// There is no concept of default worker pool after creation. So get the first worker pool that was created
+	defaultWorkerPool := workerPools[0].Name
```

Comment:
Can we blindly take zeroth element as default worker pool?
When user has multiple workerpools.. will 0th element always give default workerpool? I m not sure if we can directly take zeroth element..

Try creating a cluster and additional workerpool to a cluster.. will second apply show no infrastructure change?

Reply:
I will try this. The problem here is that the default worker pool is actually just the initial worker pool that gets created on a new cluster. After creation, it is just the same as any other worker pool. It is not marked as a default worker pool and there is no way of programmatically getting it. They do not store the creation time of the worker pools either so we cannot get the first created pool.

Reply:
That is what the issue is.. That is why we kept this on hold.. we will have to talk to IKS team to give a way to identify default workerpool